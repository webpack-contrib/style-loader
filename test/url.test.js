/* eslint-disable */
import path from 'path';
import webpack from './helpers/compiler';

describe('Loader', () => {
  test('URL', async () => {
    const config = {
      rules: [
        {
          test: /\.css$/,
          use: [
            { loader: path.resolve(__dirname, '../src/url') },
            {
              loader: 'css-loader',
              options: {}
            }
          ]
        }
      ]
    };

    const stats = await webpack('fixture.js', config);
    const { source } = stats.toJson().modules[1];

    expect(source).toMatchSnapshot();
  });
});
