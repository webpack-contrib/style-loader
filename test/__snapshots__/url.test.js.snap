// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader URL 1`] = `
"// Style Loader
// Adds CSS to the DOM by adding a <link> tag
import link from \\"!../../src/lib/url/index.js\\";
import href from \\"!!../../node_modules/css-loader/index.js??ref--0-1!./fixture.css\\";

link(href, {\\"hmr\\":true});

// Hot Module Replacement
if (module.hot) {
  module.hot.accept(\\"!!../../node_modules/css-loader/index.js??ref--0-1!./fixture.css\\", () => {
    link(require(\\"!!../../node_modules/css-loader/index.js??ref--0-1!./fixture.css\\"));
  });

  module.hot.dispose(() => link());
}"
`;
